dan_layer/engine_types/src/substate.rs

```diff
@@ -346,13 +341,105 @@ impl From<PublishedTemplateAddress> for SubstateId {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+#[error("Invalid substate id variant'{0}'")]
```

Comment:
```suggestion
#[error("Invalid substate ID variant '{0}'")]
```

I would be nice if we could know what variant is expected in the error message.

```rust
#[derive(Debug, thiserror::Error)]
#[error("Can not convert substate ID variant '{substate_id}'  to {expected}")]
pub struct InvalidSubstateIdVariant{ substate_id: SubstateId, expected: &'static str };
```

Reply:
@sdbondi Thank you for the comment, I made a change, please check if it satisfies project need.

Reply:
Perfect thanks